straxen/plugins/event_processing.py

```diff
@@ -222,103 +248,159 @@ def infer_dtype(self):
+            s1_latest_time = largest_s2s[0]['time']
+        else:
+            s1_latest_time = np.inf
+
+        largest_s1s, s1_idx = self.get_largest_sx_peaks(
+            peaks,
+            s_i=1,
+            s1_before_time=s1_latest_time,
+            s1_min_coincidence=self.config['event_s1_min_coincidence'])
+
+        self.set_sx_index(event, s1_idx, s2_idx)
+        self.set_event_properties(event, largest_s1s, largest_s2s, peaks)
+
+        # Loop over S1s and S2s and over main / alt.
+        for i, largest_s_i in enumerate([largest_s1s, largest_s2s]):
+            s_i = i + 1
+            # Largest index 0 -> main sx, 1 -> alt sx
+            for largest_index, main_or_alt in enumerate(['s', 'alt_s']):
+                peak_properties_to_save = [name for name, _, _ in self.peak_properties]
                 if s_i == 2:
-                    for name in posrec_save:
-                        result[f'alt_s{s_i}_{name}'] = secondary_s[s_i][name]
-                # Compute delay time properties
-                result[f'alt_s{s_i}_delay'] = (secondary_s[s_i]['center_time']
-                                               - main_s[s_i]['center_time'])
-
+                    peak_properties_to_save += ['x', 'y']
+                    peak_properties_to_save += self.posrec_save
+                field_names = [f'{main_or_alt}{s_i}_{name}' for name in peak_properties_to_save]
+                self.copy_largest_peaks_in_events(event,
+                                                  largest_s_i,
+                                                  largest_index,
+                                                  field_names,
+                                                  peak_properties_to_save)
+
+    @staticmethod
+    @numba.njit
+    def set_event_properties(result, largest_s1s, largest_s2s, peaks):
+        """Get properties like drift time and area before main S2"""
         # Compute drift times only if we have a valid S1-S2 pair
-        if len(main_s) == 2:
-            result['drift_time'] = \
-                main_s[2]['center_time'] - main_s[1]['center_time']
-            if 1 in secondary_s:
-                result['alt_s1_interaction_drift_time'] = \
-                    main_s[2]['center_time'] - secondary_s[1]['center_time']
-            if 2 in secondary_s:
-                result['alt_s2_interaction_drift_time'] = \
-                    secondary_s[2]['center_time'] - main_s[1]['center_time']
+        if len(largest_s1s) > 0 and len(largest_s2s) > 0:
+            result['drift_time'] = largest_s2s[0]['center_time'] - largest_s1s[0]['center_time']
+            if len(largest_s1s) > 1:
+                result['alt_s1_interaction_drift_time'] = largest_s2s[0]['center_time'] - largest_s1s[1]['center_time']
+            if len(largest_s2s) > 1:
+                result['alt_s2_interaction_drift_time'] = largest_s2s[1]['center_time'] - largest_s1s[0]['center_time']
```

Comment:
I am wondering why we never added alt_S2 - alt_S1. Could be interesting for multi-scatter studies is not it?

Reply:
Main reason is most likely that it's not used anywhere. One can of course just subtracy the alt s1 s2 centertimes. I'm actually not entirely convinced we need the alt_*_drift time for this exact reason